src/cpp/src/image_generation/models/autoencoder_kl.cpp

```diff
@@ -182,7 +182,18 @@ AutoencoderKL::AutoencoderKL(const std::string& vae_encoder_model,
     compile(device, *extract_adapters_from_properties(properties));
 }
 
-AutoencoderKL::AutoencoderKL(const AutoencoderKL&) = default;
+AutoencoderKL::AutoencoderKL(const AutoencoderKL& rhs) = default;
+
+AutoencoderKL AutoencoderKL::clone() {
+    // Support for cloning before compilation implemented below. TODO: Remove assertion when approved
+    //OPENVINO_ASSERT(!m_decoder_model, "AutoencoderKL must be compiled first. Cannot clone non-compiled model");
```

Comment:
Please uncomment these asserts

- For future improvements: we should add `if(m_model)` and support both cases - when the model is compiled and when it's not. If the model isn’t compiled, we should create a copy of the original model

Reply:
I'm sorry, but I don't understand this request. In previous review you wanted me to add support for cloning-before-compilation. Reverting those asserts completely removes this feature...

Reply:
I understand now, we should also clone m_model in case of cloning before compilation
use clone method of ov::Model